GoogleSignIn/Sources/Public/GoogleSignIn/GIDSignIn.h

```diff
@@ -70,15 +70,17 @@ typedef NS_ERROR_ENUM(kGIDSignInErrorDomain, GIDSignInErrorCode) {
 
 /// Configures `GIDSignIn` for use.
 ///
+/// @param useDebugProvider Whether or not App Check should use the debug provider.
 /// @param completion A nullable callback block passing back any error arising from the
-/// configuration process if any exists.
+///     configuration process if any exists.
 ///
 /// Call this method on `GIDSignIn` prior to use and as early as possible. This method generates App
 /// Attest key IDs and the attestation object eagerly to minimize latency later on during the sign
 /// in or add scopes flows.
-- (void)configureWithCompletion:(nullable void (^)(NSError * _Nullable error))completion
+- (void)configureWithDebugProvider:(BOOL)useDebugProvider
+                        completion:(nullable void (^)(NSError * _Nullable error))completion
```

Comment:
I'm not sure if this would make the API too complex but have you considered having two methods like:
```
// Calls the method below with GIDAppCheckProviderAppAttest
- (void)configureWithCompletion:(nullable void (^)(NSError * _Nullable error))completion;
- (void)configureWithProvider:(GIDAppCheckProvider)appCheckProvider
                        completion:(nullable void (^)(NSError * _Nullable error))completion;
```

```
typedef NS_ENUM(NSInteger, GIDAppCheckProvider) {
    GIDAppCheckProviderAppAttest,
    GIDAppCheckProviderDebug
};
```

As a downside it would be a mistake if a developer called both configure methods. On the plus side it would allow additional attestation providers to be supported in the future. (E.g., despite being older, DeviceCheck support could be valuable if this work is extended to macOS in the future.)

Reply:
If the above API is too complex, something else to consider would be:
```
- (void)configureWithCompletion:(nullable void (^)(NSError * _Nullable error))completion
               useDebugProvider:(BOOL)useDebugProvider
```
I don't like that the completion handler isn't at the end but I still worry that `configureWithDebugProvider` reads like you're configuring GSI to use the debug provider, despite it being a BOOL.

Reply:
I did think about how to represent which debug provider to use. I like the enum idea, but given that the enum would only have two cases (and we don't have plans to extend this work to macOS just yet), an enum is essentially a `BOOL`. Additionally, I would like to bias against adding more types to the API surface since that increases what developers have to keep in their head and use, and also increases maintenance costs (admittedly only very incrementally for this change).

Finally, if we do decide to extend our support to DeviceCheck + macOS, then moving to an enum would be a great win with little refactoring required at the time. IMHO, we can wait for that time.

Reply:
nit: I still think `configureWithDebugProvider:` may be a confusing method name for developers. Without seeing the `BOOL` data type, I'd expect to be passing in an instance of a provider to that parameter.

I'm not sure if this is too verbose but what about:
```
- (void)configureWithDebugProviderEnabled:(BOOL)useDebugProvider
                        completion:(nullable void (^)(NSError * _Nullable error))completion
  NS_SWIFT_NAME(configure(debugProviderEnabled:completion:));
```

For Swift devs it'd look something like:
```
GIDSignIn.sharedInstance.configure(debugProviderEnabled: true) { error in
   // ...
}
```
`debugModeEnabled` or just `debugEnabled` might also work.

That said, I'm not super concerned since I think the current `NS_SWIFT_NAME(configure(usingDebugProvider:completion:))` reads reasonably well and 3P devs are probably more likely to be using the Swift API than the ObjC API.

Reply:
I agree. Updated.